range.c

```diff
@@ -1233,6 +1233,8 @@ range_max(int argc, VALUE *argv, VALUE range)
 
         if (c > 0)
             return Qnil;
+        if (RB_INTEGER_TYPE_P(b) && !RB_INTEGER_TYPE_P(e))
+            return rb_call_super(argc, argv);
```

Comment:
While this will work, isn't it much slower than `e.floor if e > b`? Also, doesn't this loop forever for `e == Float::INFINITY`?

Reply:
It probably is much slower.  Even if the end is not infinite, it can probably appear to enter an infinite loop.  Using `floor` would definitely be more efficient.  Care probably needs to be taken in the exclusive range case when `e.floor == e`, though.